--- a/AUTHORS
+++ b/AUTHORS
@@ -1,2 +1,2 @@
 Maintainer
-	Pat Padgett <agroman@agroman.net>
+	Pat Padgett <pat@patpadgett.com>
--- a/README
+++ b/README
@@ -27,7 +27,7 @@ it elsewhere.
 Where Do I Get It?
 ------------------
 Corkscrew's primary distribution site is :
-http://www.agroman.net/corkscrew/ 
+https://github.com/patpadgett/corkscrew
 
 
 How Do I Build It?
@@ -90,7 +90,7 @@ if you could include the following information :
 Who Am I?
 ---------
 My name is Pat Padgett.  I'm a dork.
-URL   : http://www.agroman.net/
-Email : agroman@agroman.net
+URL   : https://github.com/patpadgett
+Email : pat@patpadgett.com
 
 This has been a traumatized production.  http://www.trauma-inc.com/
--- a/corkscrew.c
+++ b/corkscrew.c
@@ -127,7 +127,7 @@ void usage (void)
 void usage ()
 #endif
 {
-	printf("corkscrew %s (agroman@agroman.net)\n\n", VERSION);
+	printf("corkscrew %s (pat@patpadgett.com)\n\n", VERSION);
 	printf("usage: corkscrew <proxyhost> <proxyport> <desthost> <destport> [authfile]\n");
 }
 
